### PR TITLE
Check the PGPROC's validity properly

### PR DESCRIPTION
**Pull request by onderkalaci**
We used to only check whether the PID is valid
or not. However, Postgres does not necessarily
set the PID of the backend to 0 when it exists.

Instead, we need to be able to check it from procArray.
IsBackendPid() is what pg_stat_activity also relies
on for a similar purpose.

Fixes https://github.com/citusdata/citus/issues/6074

### CODE REVIEW

**Comment by onderkalaci** [comment]
@onurctirtir there are 2 commits, each approaches differently. I'll think a little more on which one is better. If you have any comments in the mean time, let me know.

**Comment by onurctirtir** [comment]
For second approach, I couldn't follow who resets `->activeBackend`, but I'm already ok with the first one.

**Comment by onderkalaci** [comment]
> For the second approach, I couldn't follow who resets ->activeBackend

pushed now, forgot to add that the file that uses it :)

File: src/backend/distributed/transaction/backend_data.c [hunk 1]
```diff
@@ -392,7 +393,7 @@ StoreAllActiveTransactions(Tuplestorestate *tupleStore, TupleDesc tupleDescripto
 
 		SpinLockAcquire(&currentBackend->mutex);
 
-		if (currentProc->pid == 0)
+		if (!IsBackendPid(currentProc->pid))
```
[the remark this answers is not in the document]
Comment (onderkalaci) [reply src/backend/distributed/transaction/backend_data.c]: Not really, the name of StoreAllActiveTransactions is too confusing, see https://github.com/citusdata/citus/issues/5283

It in fact returns all the executions irrespective of dist txes

**Comment by onurctirtir** [comment]
> NOT accurate to check the pid,

I guess this is still useful, pid == 0 at least indicates that no backends has been assigned to this entry so far

**Comment by onurctirtir** [comment]
>  I'm more inclined for the second approach, 

Given that we're pretty sure that our clean-up callback will get called at backend-exit, this should be fine I guess. As far as I remember from the discussion that we made a few days ago, I guess the only case that it doesn't get called is when the backend crashes, but this doesn't have any importance at all.

**Review by onurctirtir — APPROVED** [review]

**Pull request merged**